### DISCUSSION TITLE
JSC is supported in 2.1.0 and when will V8 be supported?

### DISCUSSION THREAD

**Discussion by hanhailong** [software-mansion/react-native-reanimated#1913] [Ideas]
JSC is supported in 2.1.0 and when will V8 be supported?

**Comment by hanhailong** [comment]
When switched to V8, the animation did not respond,and report the following error
![image](https://user-images.githubusercontent.com/4241807/113820219-f8d7e680-97ac-11eb-80f5-caa8db1aced3.png)

**Comment by jgonet** [comment]
This is duplicate of https://github.com/software-mansion/react-native-reanimated/discussions/1897.

**Comment by piaskowyk** [comment]
@hanhailong Is the specific reason why you want to use V8 engine?

**Reply by hanhailong** [reply]
@piaskowyk  in our app the only engine for RN is v8,v8 performs better than Hermes in some ways  https://docs.google.com/spreadsheets/d/1XB6fuk-NYZbCDikxQOAJemE-P8cfbsfdXivTeVwcwIk/edit?usp=sharing

**Reply by andreialecu** [reply]
@piaskowyk take a look at https://github.com/facebook/hermes/issues/495#issuecomment-817823392

It appears that v8 with jit mode is almost 15x faster than Hermes, and 7x faster than JSC in some tasks.

**Reply by terrysahaidak** [reply]
> @piaskowyk take a look at https://github.com/facebook/hermes/issues/495#issuecomment-817823392
> 
> It appears that v8 with jit mode is almost 15x faster than Hermes, and 7x faster than JSC in some tasks.

It makes sense as long as you don't plan to ever use v8 on iOS. Because JIT is not allowed by Apple.

**Reply by andreialecu** [reply]
@terrysahaidak React Native iOS performance is generally top-notch, so not much sense using it there. So you would be correct, I'd never plan to use it on iOS. :)

On Android however, it makes the difference between an unusable app and an usable one. 

For example, we have an app that shows an event list, and on Android with the Intl polyfills with Hermes, loading that specific screen is extremely laggy. Opening the screen takes several seconds, and the JS thread goes to `-2` fps.

With v8-jit that issue doesn't exist.

Unfortunately, we're already using reanimated 2 in other places, so we cannot switch to v8 because reanimated won't allow it.

**Reply by andreialecu** [reply]
Oh, and btw, regarding v8 and iOS - NativeScript is planning to switch to v8 (in jitless mode) for iOS: 
https://blog.nativescript.org/the-new-ios-runtime-powered-by-v8/
https://github.com/NativeScript/NativeScript/issues/8624

So it's not entirely unheard of.

**Comment by adkenyon** [comment]
We A/B tested V8 and hermes in our production Android app and found substantial improvements. Nearly everything we measured improved by 20% with some screens rendering nearly 60% faster.

Would love to see reanimated support for V8 on Android.

**Comment by piaskowyk** [comment]
It is possible that this PR (https://github.com/software-mansion/react-native-reanimated/pull/1960) will add support for V8 but I haven't tested it yet. I'll notify you after the tests.

**Reply by jzxchiang1** [reply]
The error you saw with `callWithThis` is fixed now. I've tried reanimated with V8 (using a patch) and it works beautifully. Official support would be awesome!

**Comment by jzxchiang1** [comment]
Bump. react-native-v8 is now compatible with reanimated v2 (I tested it myself with the help of @Kudo).

Can we please get official support from reanimated as soon as possible? Thank you!

https://github.com/software-mansion/react-native-reanimated/pull/1960#issuecomment-940043904

**Reply by piaskowyk** [reply]
Awesome news! Thanks for the update! We working now on building Reanimated from source on the client side. It will be useful to provide support for v8. After stable release, I will try again to add support for v8.

**Reply by jzxchiang1** [reply]
Any chance we can get v8 support soon? I think it is just updating a couple build files and renaming some libraries. Not totally sure, but it would make a huge difference. Thank you.
